src/js/xapp/index.js

```diff
@@ -0,0 +1,83 @@
+import { XAppTemplateParser } from "./parser.js";
+import { getProp } from "./util.js";
+
+class Template {
+    constructor() {
+        this.root = null;
+        this.eventUsed = [];
+    }
+
+    press(xStore) {
+        return this.root.press(xStore);
+    }
+
+    parse(rootDomNode) {
+        const parser = new XAppTemplateParser();
+        const result = parser.parse(rootDomNode);
+        this.eventUsed = result.eventUsed;
+        this.root = result.root;
+    }
+}
+
+export class XApp {
+    constructor(config) {
+        /** @type {Element} */
+        this.el = config.el;
+        this.template = new Template();
+        this.template.parse(this.el);
+        this.data = {};
+        this.methods = {};
+        this.elements = {};
+        this.hooks = {};
+    }
+
+    setHandler(el) {
+        this.template.eventUsed.forEach(eventType => {
+            el.addEventListener(eventType, e => {
+                const xEvents = e.target.xEvents;
+                if (Array.isArray(xEvents)) {
+                    const eventInfo = xEvents.find(evt => evt.property === eventType);
+                    if (eventInfo) {
+                        const args = eventInfo.value.split(",");
+                        const data = { ...this.data, ...eventInfo.binding };
+                        const values = args.slice(1).map(v => getProp(data, v));
+                        this.methods[args[0]].apply(this, [e, ...values]);
+                    }
+                }
+            });
```

Comment:
depth가 조금 깊은 것 같아요!!..
```suggestion
                if (!Array.isArray(xEvents)) {
                   return;
                }
                 const eventInfo = xEvents.find(evt => evt.property === eventType);
                 if (eventInfo) {
                        const args = eventInfo.value.split(",");
                        const data = { ...this.data, ...eventInfo.binding };
                        const values = args.slice(1).map(v => getProp(data, v));
                        this.methods[args[0]].apply(this, [e, ...values]);
                    }
                });
```
이런식으로 줄일 수 있을 것 같아요~